pymc_marketing/mmm/delayed_saturated_mmm.py

```diff
@@ -210,16 +313,84 @@ def _get_fourier_models_data(self) -> pd.DataFrame:
         """
         if self.yearly_seasonality is None:
             raise ValueError("yearly_seasonality must be specified.")
-
         date_data: pd.Series = pd.to_datetime(
-            arg=self.data[self.date_column], format="%Y-%m-%d"
+            arg=X[self.date_column], format="%Y-%m-%d"
         )
         periods: npt.NDArray[np.float_] = date_data.dt.dayofyear.to_numpy() / 365.25
         return generate_fourier_modes(
             periods=periods,
             n_order=self.yearly_seasonality,
         )
 
+    @property
+    def _serializable_model_config(self) -> Dict[str, Any]:
+        serializable_config = self.model_config.copy()
+        return serializable_config
+
+    def _data_setter(
+        self,
+        X: Union[np.ndarray, pd.DataFrame],
+        y: Optional[Union[np.ndarray, pd.Series]] = None,
+    ) -> None:
+        """
+        Sets new data in the model.
+
+        This function accepts data in various formats and sets them into the
+        model using the PyMC's `set_data` method. The data corresponds to the
+        channel data and the target.
+
+        Parameters
+        ----------
+        X : Union[np.ndarray, pd.DataFrame]
+            Data for the channel. It can be a numpy array or pandas DataFrame.
+            If it's a DataFrame, the columns corresponding to self.channel_columns
+            are used. If it's an ndarray, it's used directly.
+        y : Union[np.ndarray, pd.Series], optional
+            Target data. It can be a numpy array or a pandas Series.
+            If it's a Series, its values are used. If it's an ndarray, it's used
+            directly. The default is None.
+
+        Raises
+        ------
+        RuntimeError
+            If the data for the channel is not provided in `X`.
+        TypeError
+            If `X` is not a pandas DataFrame or a numpy array, or
+            if `y` is not a pandas Series or a numpy array and is not None.
+
+        Returns
+        -------
+        None
```

Comment:
I think we can also remove these None returns

Reply:
Most of the docstrings are auto-generated by chatGPT, I'm doing an experiment where I try to see how much of the boring stuff can I outsource to an AI. Obviously it needs some supervision, but as I didn't see this stage of PR as the final one I didn't review every single one of them